contracts/rollup/BatchManager.sol

```diff
@@ -127,14 +121,17 @@ contract BatchManager is Parameters {
                 block.number + paramBlocksToFinalise
             )
         });
-        batches.push(newBatch);
-        emit NewBatch(msg.sender, batches.length - 1, batchType);
+        emit NewBatch(msg.sender, nextBatchID, batchType);
+        nextBatchID++;
     }
 
     /**
      * @notice Withdraw delay allows coordinators to withdraw their stake after the batch has been finalised
      */
     function withdrawStake(uint256 batchID) public {
+        // A valid batch should have non-zero meta
+        // We use this to check if a stake has been withdrawn
+        require(batches[batchID].meta != bytes32(0), "Rollup: No such batch");
```

Comment:
The meta has fields like batchType, commitmentLength, committer, finaliseOn. The last two should guarantee the existent batch have a non-zero meta.